cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -1085,9 +1091,6 @@ public boolean apply(co.cask.cdap.proto.id.ArtifactId input) {
     };
 
     for (Map.Entry<byte[], byte[]> column : columns.entrySet()) {
-      if (limit != Integer.MAX_VALUE && limit == plugins.size()) {
```

Comment:
instead of removing all the logic, should fix this method so that after something is added to the map, if the size is greater than the limit, remove an item from the map.